primitives/core/src/crypto.rs

```diff
@@ -553,7 +553,7 @@ pub trait Public: AsRef<[u8]> + AsMut<[u8]> + Default + Derive + CryptoType + Pa
 }
 
 /// An opaque 32-byte cryptographic identifier.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Default, Encode, Decode)]
+#[derive(Clone, Hash, Eq, PartialEq, Ord, PartialOrd, Default, Encode, Decode)]
```

Comment:
```suggestion
#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Default, Encode, Decode)]
#[cfg_attr(features="std", derive(Hash))]
```

if it is "not in runtime"

Reply:
Hash is in std::core? https://doc.rust-lang.org/core/hash/trait.Hash.html

Reply:
Our "std" feature is for things which don't compile to the runtime. We don't need this `Hash` derivation compiled to the runtime.

Reply:
Ah, I see. Yeah makes sense then. I also potentially wanted to add a `From<sr25519::Public>` implementation (as well perhaps for the other sp_core key types to be complete). Would that be welcome?